dali/operators/generic/reshape.cc

```diff
@@ -197,32 +198,77 @@ bool Reshape<Backend>::SetupImpl(std::vector<OutputDesc> &output_desc, const Wor
   return false;
 }
 
+template <bool relative, typename OutSampleShape, typename InSampleShape, typename ShapeArg>
+void SetOutputSampleShape(OutSampleShape &&out_shape,
+                          const InSampleShape &in_shape,
+                          const ShapeArg &arg,
+                          const int *src_dims,
+                          int out_ndim) {
+  for (int d = 0; d < out_ndim; d++) {
+    auto e = arg[d];
+
+    int out_e = 1;
+    if (e < 0) {
+      out_e = -1;
+    } else if (relative) {
+      int src_d = src_dims ? src_dims[d] : d;
+      out_e = src_d < 0 ? 1 : round_int(e * in_shape[src_d]);
+    } else {
+      out_e = e;
+    }
+
+    out_shape[d] = out_e;
+  }
+}
+
+template <typename Backend>
+void Reshape<Backend>::ValidateRelativeShapeNDim(int ndim) const {
+  if (use_src_dims_) {
+    if (ndim + 0_uz != src_dims_.size()) {
```

Comment:
Is it to avoid cast?

Reply:
In the beginning it was a part of a longer expression and the `static_cast<size_t>` called for a line break and made it hardly readable. Now I could restore it if you insist.